lib/proxy/plugin.rb

```diff
@@ -15,59 +16,53 @@ def initialize(aname, aversion)
 end
 
 class ::Proxy::Plugins
-  @@loaded = [] # {:name, :version, :class, :factory}
-  @@enabled = {} # plugin_name => instance
+  extend ::Proxy::Log
 
   class << self
     def plugin_loaded(a_name, a_version, a_class)
-      @@loaded += [{:name => a_name, :version => a_version, :class => a_class}]
+      self.loaded += [{:name => a_name, :version => a_version, :class => a_class, :enabled => false}]
     end
 
-    def configure_loaded_plugins
-      configuration_order = build_configuration_order(@@loaded)
-      configuration_order.each { |plugin| plugin[:class].new.configure_plugin }
+    def loaded
+      @loaded ||= [] # {:name, :version, :class, :factory, :instance, :enabled}
     end
 
-    def build_configuration_order(loaded_plugins)
-      plugins_only = loaded_plugins.select {|plugin| plugin[:class].ancestors.include?(::Proxy::Plugin)}
-      configuration_order = []
-
-      # FIX ME: config order currently is unaffected by provider prerequisites
-      plugins_only.each do |plugin|
-        next if configuration_order.include?(plugin)
-        prerequisite_names = plugin[:class].initialize_after - configuration_order.map { |p| p[:name] }
-        prerequisites = loaded_plugins.select {|p| prerequisite_names.include?(p[:name])}
-        configuration_order = configuration_order + prerequisites + [plugin]
-      end
-
-      configuration_order
+    def loaded=(an_array)
+      @loaded = an_array
     end
 
-    def plugin_enabled(plugin_name, instance)
-      @@enabled[plugin_name.to_sym] = instance
+    def update(updated_plugins)
+      updated_plugins.each do |updated|
+        loaded.delete_if {|p| p[:name] == updated[:name]}
+        loaded << updated
+      end
     end
 
     def plugin_enabled?(plugin_name)
-      !!@@enabled[plugin_name.to_sym]
+      plugin = loaded.find {|p| p[:name] == plugin_name.to_sym}
+      plugin.nil? ? false : !!plugin[:enabled]
     end
 
     def disable_plugin(plugin_name)
-      @@enabled.delete(plugin_name.to_sym)
+      logger.error("::Proxy::Plugins.disable_plugin has been deprecated and will be removed from future versions of smart-proxy")
```

Comment:
1. warn, not error
2. suggest raising an exception as the preferred alternative